Flaky Test [Initializing / Test / windows-3.8-none / test_send_remote_failover_sync[validating_httpserver0-sending_elasticapm_client0] – tests.client.client_tests]

## Flaky Test

* **Test Name:** `Initializing / Test / windows-3.8-none / test_send_remote_failover_sync[validating_httpserver0-sending_elasticapm_client0] – tests.client.client_tests`
* **Artifact Link:** https://apm-ci.elastic.co/blue/organizations/jenkins/apm-agent-python%2Fapm-agent-python-mbp%2FPR-824/detail/PR-824/161/
* **PR:** #824
* **Commit:** d16b3cc19033da48037cb31b1ef44d5decb690de


### Error details
```
AssertionError
```


### Stack Trace
```
should_try = <MagicMock name="should_try" id="2387705179872">
sending_elasticapm_client = <elasticapm.base.Client object at 0x0000022BEE657BE0>
caplog = <_pytest.logging.LogCaptureFixture object at 0x0000022BEE51A940>

    @pytest.mark.flaky(reruns=3)  # test is flaky on Windows
    @pytest.mark.parametrize(
        "sending_elasticapm_client",
        [{"transport_class": "elasticapm.transport.http.Transport", "async_mode": False}],
        indirect=True,
    )
    @pytest.mark.parametrize("validating_httpserver", [{"app": ContentServer}], indirect=True)
    @mock.patch("elasticapm.transport.base.TransportState.should_try")
    def test_send_remote_failover_sync(should_try, sending_elasticapm_client, caplog):
        sending_elasticapm_client.httpserver.code = 400
        sending_elasticapm_client.httpserver.content = "go away"
        should_try.return_value = True
    
        # test error
        with caplog.at_level("ERROR", "elasticapm.transport"):
            sending_elasticapm_client.capture_message("foo", handled=False)
        sending_elasticapm_client._transport.flush()
        assert sending_elasticapm_client._transport.state.did_fail()
>       assert_any_record_contains(caplog.records, "go away")

tests\client\client_tests.py:253: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

records = [<LogRecord: urllib3.connectionpool, 30, C:\Users\jenkins\workspace\thon_apm-agent-python-mbp_PR-824\src\github.com\el...-mbp_PR-824\src\github.com\elastic\apm-agent-python\elasticapm\transport\base.py, 297, "Failed to submit message: %r">]
message = "go away", logger = None

    def assert_any_record_contains(records, message, logger=None):
>       assert any(
            message in record.message
            for record in (record for record in records if logger is None or record.name == logger)
        )
E       AssertionError

tests\utils\__init__.py:33: AssertionError
```

## Flaky Test

* **Test Name:** `Initializing / Test / windows-3.8-none / test_send_remote_failover_sync[validating_httpserver0-sending_elasticapm_client0] – tests.client.client_tests`
* **Artifact Link:** https://apm-ci.elastic.co/blue/organizations/jenkins/apm-agent-python%2Fapm-agent-python-mbp%2FPR-1308/detail/PR-1308/6/
* **PR:** #1308
* **Commit:** d74b27e2c5fc2480030d0f08fb08bdbff0d1aea8


### Error details
```
AssertionError
```


### Stack Trace
```
should_try = <MagicMock name="should_try" id="1682332568352">
sending_elasticapm_client = <elasticapm.base.Client object at 0x00000187B2DA9B50>
caplog = <_pytest.logging.LogCaptureFixture object at 0x00000187B2D5CBE0>

    @pytest.mark.flaky(reruns=3)  # test is flaky on Windows
    @pytest.mark.parametrize(
        "sending_elasticapm_client",
        [{"transport_class": "elasticapm.transport.http.Transport", "async_mode": False}],
        indirect=True,
    )
    @pytest.mark.parametrize("validating_httpserver", [{"app": ContentServer}], indirect=True)
    @mock.patch("elasticapm.transport.base.TransportState.should_try")
    def test_send_remote_failover_sync(should_try, sending_elasticapm_client, caplog):
        sending_elasticapm_client.httpserver.code = 400
        sending_elasticapm_client.httpserver.content = "go away"
        should_try.return_value = True
    
        # test error
        with caplog.at_level("ERROR", "elasticapm.transport"):
            sending_elasticapm_client.capture_message("foo", handled=False)
        sending_elasticapm_client._transport.flush()
        assert sending_elasticapm_client._transport.state.did_fail()
>       assert_any_record_contains(caplog.records, "go away")

tests\client\client_tests.py:253: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

records = [<LogRecord: urllib3.connectionpool, 30, C:\Users\jenkins\workspace\hon_apm-agent-python-mbp_PR-1308\src\github.com\el...mbp_PR-1308\src\github.com\elastic\apm-agent-python\elasticapm\transport\base.py, 297, "Failed to submit message: %r">]
message = "go away", logger = None

    def assert_any_record_contains(records, message, logger=None):
>       assert any(
            message in record.message
            for record in (record for record in records if logger is None or record.name == logger)
        )
E       AssertionError

tests\utils\__init__.py:33: AssertionError
```

## Flaky Test

* **Test Name:** `Initializing / Test / windows-3.8-none / test_send_remote_failover_sync[validating_httpserver0-sending_elasticapm_client0] – tests.client.client_tests`
* **Artifact Link:** https://apm-ci.elastic.co/blue/organizations/jenkins/apm-agent-python%2Fapm-agent-python-mbp%2Fmaster/detail/master/518/
* **PR:** None
* **Commit:** 6ea576942a3bf56d8d930c827195db634c22271a


### Error details
```
AssertionError
```


### Stack Trace
```
should_try = <MagicMock name="should_try" id="2453634977744">
sending_elasticapm_client = <elasticapm.base.Client object at 0x0000023B48032DF0>
caplog = <_pytest.logging.LogCaptureFixture object at 0x0000023B480A8A60>

    @pytest.mark.flaky(reruns=3)  # test is flaky on Windows
    @pytest.mark.parametrize(
        "sending_elasticapm_client",
        [{"transport_class": "elasticapm.transport.http.Transport", "async_mode": False}],
        indirect=True,
    )
    @pytest.mark.parametrize("validating_httpserver", [{"app": ContentServer}], indirect=True)
    @mock.patch("elasticapm.transport.base.TransportState.should_try")
    def test_send_remote_failover_sync(should_try, sending_elasticapm_client, caplog):
        sending_elasticapm_client.httpserver.code = 400
        sending_elasticapm_client.httpserver.content = "go away"
        should_try.return_value = True
    
        # test error
        with caplog.at_level("ERROR", "elasticapm.transport"):
            sending_elasticapm_client.capture_message("foo", handled=False)
        sending_elasticapm_client._transport.flush()
        assert sending_elasticapm_client._transport.state.did_fail()
>       assert_any_record_contains(caplog.records, "go away")

tests\client\client_tests.py:253: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

records = [<LogRecord: urllib3.connectionpool, 30, C:\Users\jenkins\workspace\thon_apm-agent-python-mbp_master\src\github.com\el...-mbp_master\src\github.com\elastic\apm-agent-python\elasticapm\transport\base.py, 297, "Failed to submit message: %r">]
message = "go away", logger = None

    def assert_any_record_contains(records, message, logger=None):
>       assert any(
            message in record.message
            for record in (record for record in records if logger is None or record.name == logger)
        )
E       AssertionError

tests\utils\__init__.py:33: AssertionError
```

## Flaky Test

* **Test Name:** `Initializing / Test / windows-3.8-none / test_send_remote_failover_sync[validating_httpserver0-sending_elasticapm_client0] – tests.client.client_tests`
* **Artifact Link:** https://apm-ci.elastic.co/blue/organizations/jenkins/apm-agent-python%2Fapm-agent-python-mbp%2Fmaster/detail/master/522/
* **PR:** None
* **Commit:** 6f295d263ccb96c4127dfdee1c146d5838a0533b


### Error details
```
AssertionError
```


### Stack Trace
```
should_try = <MagicMock name="should_try" id="1947020828384">
sending_elasticapm_client = <elasticapm.base.Client object at 0x000001C553908C10>
caplog = <_pytest.logging.LogCaptureFixture object at 0x000001C5537C1940>

    @pytest.mark.flaky(reruns=3)  # test is flaky on Windows
    @pytest.mark.parametrize(
        "sending_elasticapm_client",
        [{"transport_class": "elasticapm.transport.http.Transport", "async_mode": False}],
        indirect=True,
    )
    @pytest.mark.parametrize("validating_httpserver", [{"app": ContentServer}], indirect=True)
    @mock.patch("elasticapm.transport.base.TransportState.should_try")
    def test_send_remote_failover_sync(should_try, sending_elasticapm_client, caplog):
        sending_elasticapm_client.httpserver.code = 400
        sending_elasticapm_client.httpserver.content = "go away"
        should_try.return_value = True
    
        # test error
        with caplog.at_level("ERROR", "elasticapm.transport"):
            sending_elasticapm_client.capture_message("foo", handled=False)
        sending_elasticapm_client._transport.flush()
        assert sending_elasticapm_client._transport.state.did_fail()
>       assert_any_record_contains(caplog.records, "go away")

tests\client\client_tests.py:253: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

records = [<LogRecord: urllib3.connectionpool, 30, C:\Users\jenkins\workspace\thon_apm-agent-python-mbp_master\src\github.com\el...-mbp_master\src\github.com\elastic\apm-agent-python\elasticapm\transport\base.py, 297, "Failed to submit message: %r">]
message = "go away", logger = None

    def assert_any_record_contains(records, message, logger=None):
>       assert any(
            message in record.message
            for record in (record for record in records if logger is None or record.name == logger)
        )
E       AssertionError

tests\utils\__init__.py:33: AssertionError
```

--- a/tests/client/client_tests.py
+++ b/tests/client/client_tests.py
@@ -232,7 +232,7 @@ def test_empty_processor_list(elasticapm_client):
     assert elasticapm_client.processors == []
 
 
-@pytest.mark.flaky(reruns=3)  # test is flaky on Windows
+@pytest.mark.skipif(platform.system() == "Windows", reason="Flaky test on windows")
 @pytest.mark.parametrize(
     "sending_elasticapm_client",
     [{"transport_class": "elasticapm.transport.http.Transport", "async_mode": False}],

